src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/PetHotel.java

```diff
@@ -0,0 +1,39 @@
+package serenitylabs.tutorials.vetclinic.collections.katas;
+
+import serenitylabs.tutorials.vetclinic.Pet;
+
+import java.util.*;
+
+import static java.util.Comparator.comparing;
+
+/**
+ * Created by vdheer on 9/20/2016.
+ */
+public class PetHotel {
+    private final Collection<Pet> registeredPets = new TreeSet<>(comparing(Pet::getName));
+    private final Queue<Pet> waitingListOfPets = new LinkedList<>();
+    private static final int DEFAULT_MAXIMUM_CAPACITY = 20;
+
+    private static String hotelName;
+
+    public List<Pet> getRegisteredPets() {
+        return new ArrayList<>(registeredPets);
+    }
+
+    public BookingResponse checkIn(Pet pet) {
+        return (registeredPets.size() < DEFAULT_MAXIMUM_CAPACITY)
+                ? new BookingResponse(registeredPets.add(pet))
+                : new BookingResponse(waitingListOfPets.add(pet));
```

Comment:
This will return a positive booking response in both cases, so there is no way for the customer to know if the pet was successfully booked or not.